test/e2e/autoscaling/cluster_size_autoscaling.go

```diff
@@ -795,6 +795,53 @@ var _ = SIGDescribe("Cluster size autoscaling [Slow]", func() {
 		Expect(getNAPNodePoolsNumber()).Should(Equal(0))
 		framework.ExpectNoError(disableAutoprovisioning())
 	})
+
+	It("shouldn't add new node group if not needed [Feature:ClusterSizeAutoscalingScaleWithNAP]", func() {
+		framework.SkipUnlessProviderIs("gke")
+		By("Decrease size of groups")
+		nodesCountAfterResize := 0
+		for mig, size := range originalSizes {
+			if size > 1 {
+				framework.ExpectNoError(framework.ResizeGroup(mig, int32(1)))
+				nodesCountAfterResize++
+			}
+		}
+		framework.ExpectNoError(framework.WaitForReadyNodes(c, nodesCountAfterResize, resizeTimeout))
+		Expect(getNAPNodePoolsNumber()).Should(Equal(0))
+		framework.ExpectNoError(enableAutoprovisioning(""))
+		By("Create pods")
+		ReserveMemory(f, "memory-reservation", nodesCountAfterResize+1, int(float64(nodesCountAfterResize+1)*float64(0.7)*float64(memAllocatableMb)), true, scaleUpTimeout)
+		defer framework.DeleteRCAndPods(f.ClientSet, f.InternalClientset, f.Namespace.Name, "memory-reservation")
+		By("Waiting for scale up")
+		// Verify that cluster size increased.
+		framework.ExpectNoError(WaitForClusterSizeFunc(f.ClientSet,
+			func(size int) bool { return size == nodesCountAfterResize+1 }, scaleUpTimeout))
+		By("Check if NAP group was created hoping id didn't happen")
+		Expect(getNAPNodePoolsNumber()).Should(Equal(0))
+		framework.ExpectNoError(disableAutoprovisioning())
+	})
+
+	It("shouldn't scale up if cores limit too low, should scale up after limit is changed [Feature:ClusterSizeAutoscalingScaleWithNAP]", func() {
+		framework.SkipUnlessProviderIs("gke")
+		Expect(getNAPNodePoolsNumber()).Should(Equal(0))
+		framework.ExpectNoError(enableAutoprovisioning("\"resource_limits\":{\"name\":\"cpu\", \"minimum\":2, \"maximum\":3}"))
+		ReserveMemory(f, "memory-reservation", 1, int(1.1*float64(memAllocatableMb)), false, time.Second)
+		defer framework.DeleteRCAndPods(f.ClientSet, f.InternalClientset, f.Namespace.Name, "memory-reservation")
+		By(fmt.Sprintf("Waiting for scale up hoping it won't happen, sleep for %s", scaleUpTimeout.String()))
+		time.Sleep(scaleUpTimeout)
+		// Verify that cluster size is not changed
+		framework.ExpectNoError(WaitForClusterSizeFunc(f.ClientSet,
+			func(size int) bool { return size == nodeCount }, time.Second))
+		By("Change resource limits")
+		framework.ExpectNoError(enableAutoprovisioning("\"resource_limits\":{\"name\":\"cpu\", \"minimum\":2, \"maximum\":8}"))
+		By("Wait for scale up")
+		framework.ExpectNoError(WaitForClusterSizeFunc(f.ClientSet,
+			func(size int) bool { return size == nodeCount+1 }, scaleUpTimeout))
+		By("Check if NAP group was created")
+		Expect(getNAPNodePoolsNumber()).Should(Equal(1))
```

Comment:
Are we sure this should always be exactly 1 new node pool?

Reply:
+1
There shouldn't be more than one if there's only one 1 replica, but there may be none if memory request was set too low (i.e. there was a node group with larger nodes available for scale up.)

Reply:
Yes, I'm sure as there should be only one new node added.